zh_cn/news/_posts/2025-12-18-ruby-4-0-0-preview3-released.md

```diff
@@ -0,0 +1,471 @@
+        Ractor.new port1, port2 do |port1, port2|
+          port1 << 1
+          port2 << 11
+          port1 << 2
+          port2 << 12
+        end
+        2.times{ p port1.receive } #=> 1, 2
+        2.times{ p port2.receive } #=> 11, 12
+        ```
+
+        `Ractor::Port` 提供以下方法：
+
+        * `Ractor::Port#receive`
+        * `Ractor::Port#send` (或 `Ractor::Port#<<`)
+        * `Ractor::Port#close`
+        * `Ractor::Port#closed?`
+
+        由此，`Ractor.yield` 和 `Ractor#take` 被移除。
+
+    * 新增 `Ractor#join` 和 `Ractor#value`，用于等待 Ractor 终止。
+      这类似于 `Thread#join` 和 `Thread#value`。
+
+    * 新增 `Ractor#monitor` 和 `Ractor#unmonitor` 作为内部使用的底层接口来实现 `Ractor#join`。
+
+    * `Ractor.select` 现在只接受 Ractors 和 Ports 作为参数。
+      如果参数是 Ractors，方法会在某个 Ractor 终止时返回。
+
+    * 新增 `Ractor#default_port`。每个 `Ractor` 都有一个默认端口，用于
+      `Ractor.send`，`Ractor.receive`。
+
+    * 移除 `Ractor#close_incoming` 和 `Ractor#close_outgoing`。
+
+    * 引入 `Ractor.shareable_proc` 和 `Ractor.shareable_lambda` 实现可共享的 Proc 或 lambda。
+      [[Feature #21550]], [[Feature #21557]]
+
+* Range
+
+    * `Range#to_set` 和 `Enumerator#to_set` 现在会先检查大小，来避免无上限 range 的问题。
+      [[Bug #21654]]
+
+    * `Range#overlap?` 现在可以正确处理无限（无界）  range。
+      [[Bug #21185]]
+
+    * `Range#max` 在无下限整数 range 中的行为已修复。
+      [[Bug #21174]] [[Bug #21175]]
+
+* Ruby
+
+    * 定义了新的顶级模块 `Ruby`，包含了 Ruby 相关的常量，此模块在 Ruby 3.4 中被保留，
+      现在被正式定义。 [[Feature #20884]]
+
+* Ruby::Box
+
+    * 这是一项新的（实验性）功能，用于实现定义分离。
+      关于 "Ruby Box" 的详细信息，请参阅 [doc/language/box.md](https://github.com/ruby/ruby/blob/master/doc/language/box.md)。
+      [[Feature #21311]] [[Misc #21385]]
+
+* Set
+
+    * `Set` 现在是一个核心类，不再是自动加载的标准库类。
+      [[Feature #21216]]
+
+    * `Set#inspect` 现在返回一个更适合 `eval` 的字符串，
+      使用 `Set[]` 语法（如： `Set[1, 2, 3]`，代替原来的 `#<Set: {1, 2, 3}>`）。
+      这使得 `Set` 的行为能与其他核心集合类（如： `Array` 和 `Hash`）保持一致。
+      [[Feature #21389]]
+
+    * 声明废弃向方法 `Set#to_set` 和 `Enumerable#to_set` 传递参数的行为。
+      [[Feature #21390]]
+
+* Socket
+
+    * `Socket.tcp` 和 `TCPSocket.new` 现在能接受关键词参数 `open_timeout`
+      来定义初始连接时的超时时间。 [[Feature #21347]]
+
+* String
+
+    * 更新 Unicode 版本至 17.0.0，更新 Emoji 版本至 17.0。
+      [[Feature #19908]][[Feature #20724]][[Feature #21275]] （同样适用于 Regexp）
+
+    * 扩展 `String#strip`，`strip!`，`lstrip`，`lstrip!`，`rstrip` 和 `rstrip!`
+      以接受 `*selectors` 参数。 [[Feature #21552]]
+
+* Thread
+
+    * 引入和 `Kernel#raise` 类似的参数支持 `Thread#raise(cause:)`。
+      [[Feature #21360]]
+
+* Fiber
+
+    * 引入和 `Kernel#raise` 类似的参数支持 `Fiber#raise(cause:)`。
+      [[Feature #21360]]
+
+* Fiber::Scheduler
+
+    * 引入 `Fiber::Scheduler#fiber_interrupt`，允许通过指定异常来终端纤程。
+      典型的用例是当 IO 操作关闭时，中断等待此 IO 的纤程。
+      [[Feature #21166]]
+
+* Pathname
+
+    * Pathname 从默认 gem 提升为 Ruby 核心类。
+      [[Feature #17473]]
+
+## 标准库更新
+
+以下默认 gems 提升为绑定 gem。
+
+* ostruct 0.6.3
+* pstore 0.2.0
+* benchmark 0.5.0
+* logger 1.7.0
+* rdoc 6.17.0
+* win32ole 1.9.2
+* irb 1.15.3
+* reline 0.6.3
+* readline 0.0.4
+* fiddle 1.1.8
```

Comment:
Some of the stdlib update versions are incorrect.  I've corrected the `en`, `ja`, and `uk` locales in #3735.  You may want to update this PR with the correct versions.

Reply:
Well, I'll update it after #3735 is merged.

Reply:
I've updated #3735 to include your `zh_cn` translation.  Thanks!